src/Disks/IO/createReadBufferFromFileBase.cpp

```diff
@@ -80,6 +82,19 @@ std::unique_ptr<ReadBufferFromFileBase> createReadBufferFromFileBase(
         {
             res = std::make_unique<ReadBufferFromFilePReadWithDescriptorsCache>(filename, buffer_size, actual_flags, existing_memory, buffer_alignment, file_size);
         }
+        else if (settings.local_fs_method == LocalFSReadMethod::io_uring)
+        {
+#if defined(OS_LINUX)
+            static std::shared_ptr<IOUringReader> reader = std::make_shared<IOUringReader>(512);
```

Comment:
It is not possible to join the thread with static variable, this should be moved to global context, like other things (like thread pool for reading and so on).